app/extend_block.go

```diff
@@ -0,0 +1,28 @@
+package app
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	"github.com/celestiaorg/celestia-app/pkg/square"
+	"github.com/celestiaorg/rsmt2d"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+// ExtendBlock extends the given block data into a data square for a given app
+// version.
+func ExtendBlock(data coretypes.Data, appVersion uint64) (*rsmt2d.ExtendedDataSquare, error) {
+	// Construct the data square from the block's transactions
+	dataSquare, err := square.Construct(data.Txs.ToSliceOfBytes(), appVersion, appconsts.SquareSizeUpperBound(appVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	return da.ExtendShares(shares.ToBytes(dataSquare))
+}
+
+// EmptyBlock returns true if the given block data is considered empty by the
+// application at a given version.
```

Comment:
As the version is not used, maybe we can remove it from the func description as well. 

```suggestion
// EmptyBlock returns true if the given block data is considered empty.
```

Reply:
gahh dang, will fix in a different PR for main